### DISCUSSION TITLE
Support for zed editor

### DISCUSSION THREAD

**Discussion by sxwebdev** [a-h/templ#556] [General]
How to integrate templ with [zed.dev](https://zed.dev) editor ?

**Comment by a-h** [comment]
From reading the docs, it looks like it has LSP and tree sitter support, but I haven't tried out zed myself.

Templ's got an LSP, and there's a tree sitter config.

I've never used zed, so don't know how you would connect the dots though because the docs are very light.

**Reply by joerdav** [reply]
I believe at the moment language support has to be coded into the zed editor, they recently added extensions but it only supports syntax highlights. 

See more here: https://github.com/zed-industries/zed/issues/7351

**Comment by the-goodies** [comment]
Sorry to hijack, but what about Sublime Text? Sublime Text supports LSP as seen here: https://github.com/sublimelsp/LSP, and there is Go LSP implementation available at https://github.com/sublimelsp/LSP-gopls/. Would adding support for Templ in Sublime Text be a significant undertaking?

**Comment by joerdav** [comment]
Looks like this was resolved from a syntax highlighting perspective in https://github.com/zed-industries/extensions/pull/232 we just need lsp support now!
